### PR TITLE
Rubocop updates and execution

### PR DESCRIPTION
**Pull request by ObiWanKeoni**
1. Updates `rubocop.yml` to match https://github.com/rgeo/rgeo/blob/main/.rubocop.yml
2. Ran `rubocop -A` to autocorrect files

Final output of `rubocop -A`:
```bash
❯ rubocop -A
Inspecting 14 files
..............

14 files inspected, no offenses detected
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@
 Makefile*
 mkmf.log
 /Gemfile.lock
+/.idea
```
Comment (ObiWanKeoni) [inline .gitignore:16]: First time a contributor uses RubyMine I see

File: test/missing_dbf_columns_test.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def test_missing_columns
   def _open_shapefile(name_, &block_)
     RGeo::Shapefile::Reader.open(
       File.expand_path("missing_dbf_columns/#{name_}",
-      File.dirname(__FILE__)),
+                       File.dirname(__FILE__)),
```
Comment (ObiWanKeoni) [inline test/missing_dbf_columns_test.rb:20]: Interesting that rubocop doesn't "fix" this to use `__dir__` 🤔

**Review by ObiWanKeoni — COMMENTED** [review]
@keithdoggett @BuonOmo When y'all get a chance

File: .gitignore (hunk 1 above)
Comment (BuonOmo) [reply .gitignore:16]: @ObiWanKeoni no, it is just that you didn't configure your global gitignore 😉 

There are three levels of ignoring in git:

1. `.gitignore`, corresponding to what the project want to ignore
2. `.git/info/exclude`, local to a projet and a specific person
3. `~/.gitignore` (or other location configurable) that is computer wide.

I think `/.idea` belongs to the third, and not this project!

File: test/missing_dbf_columns_test.rb (hunk 2 above)
Comment (BuonOmo) [reply test/missing_dbf_columns_test.rb:20]: It use to do it in my company, I think it is a matter of configuration :)

File: rgeo-shapefile.gemspec [hunk 3]
```diff
@@ -22,4 +22,5 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "minitest", "~> 5.3"
   spec.add_development_dependency "rake", "~> 12.0"
+  spec.metadata["rubygems_mfa_required"] = "true"
```
Comment (BuonOmo) [inline rgeo-shapefile.gemspec:25]: I would rather put this with the first block (e.g. after `spec.license` for instance)

File: .rubocop.yml [hunk 4]
```diff
@@ -1,13 +1,70 @@
 inherit_from: .rubocop_todo.yml
 
+AllCops:
+  TargetRubyVersion: 2.7
```
Comment (BuonOmo) [inline .rubocop.yml:4]: Shouldn't we target ruby 3.something now ? :)

**Comment by BuonOmo** [comment]
Thank you!

I made a few comments, and I also think it would be nice to add this to the CI, WDYT?

There's already an example of this in rgeo (https://github.com/rgeo/rgeo/blob/main/.github/workflows/CI.yml) and then we'll configure it to be mandatory in PRs :)

File: .gitignore (hunk 1 above)
Comment (ObiWanKeoni) [reply .gitignore:16]: I learn something new about git every week it seems like lol

thanks for the context, didn't realize this was a thing!

File: test/missing_dbf_columns_test.rb (hunk 2 above)
Comment (ObiWanKeoni) [reply test/missing_dbf_columns_test.rb:20]: I think something is weird with my dev setup - this cop doesn't seem to be working 🤔
Any ideas?

File: .rubocop.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 inherit_from: .rubocop_todo.yml
 
 AllCops:
-  TargetRubyVersion: 2.7
+  TargetRubyVersion: 3.0
```
Comment (BuonOmo) [inline .rubocop.yml:4]: I think 3.2 is the latest, why not going all the way ? :)

File: test/missing_dbf_columns_test.rb (hunk 2 above)
Comment (BuonOmo) [reply test/missing_dbf_columns_test.rb:20]: Currently looking at that.

First thing, I think we should add rubocop to dev dependencies!

Then:

```
Style/GlobalVars:
  Exclude:
    - 'ext/geos_c_impl/extconf.rb'
 ```
 
 makes no sense here :)
 
 And finally it seems like the rule takes into account a slightly different case than ours. I'd be in favor of searching and replacing, and that's it :)

**Comment by ObiWanKeoni** [comment]
Apologies for neglecting this this week. Had a bit of a rough week. Will try to wrap this up next week!

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -1,5 +1,10 @@
-name: Tests
-on: [push, pull_request]
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (BuonOmo) [inline .github/workflows/CI.yml:7]: I'd add as well:

```suggestion
  pull_request:
  # Allow running manually the CI workflow.
  workflow_dispatch:

# This allows a subsequently queued workflow run to interrupt previous runs.
concurrency:
  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
  cancel-in-progress: true
```

found recently in gh help and applying to most of my projects!

**Comment by ObiWanKeoni** [comment]
That... was an accident. Saw shiny button, clicked shiny button lol

**Comment by ObiWanKeoni** [comment]
> If so, just send me an email at buonomo dot ulysse at gmail dot com so I have yours.

Sent by the way :)

**Review by BuonOmo — APPROVED** [review]
LGTM! @keithdoggett ?

**Review by keithdoggett — APPROVED** [review]
LGTM

**Comment by keithdoggett** [comment]
@BuonOmo do you want to squash this?

**Pull request merged**